Issues with Procedural Goal mapper generation

Discovered with @AaronPlave and @mattdailis while testing procedural goals on the [aerie-orbiter-model](https://github.com/NASA-AMMOS/aerie-orbiter-model) 

1. If you update the model to PlanDev 4.0.0 and try to run ` ./gradlew scheduling:buildAllProcedureJars` it fails with the following:
```
    /Users/delany/Documents/jpl/repos/aerie-orbiter-model/scheduling/build/generated/sources/annotationProcessor/java/main/scheduling/generated/procedures/AddPeriapsesMapper.java:95: error: cannot find symbol
      arguments.put("VERSIONED_KERNELS_ROOT_DIRECTORY", this.mapper_VERSIONED_KERNELS_ROOT_DIRECTORY.serializeValue(input.VERSIONED_KERNELS_ROOT_DIRECTORY()));
                                                                                                                     ^
  symbol:   method VERSIONED_KERNELS_ROOT_DIRECTORY()
  location: variable input of type AddPeriapses
```
2. It now generates files called eg. `AddApoapsisMapper.jar` instead of `AddApoapsis.jar` like before

The first issue (we believe) is fixed by changing [this line of ProcedureProcessor](https://github.com/NASA-AMMOS/aerie/blob/e4b9061397bbec62fd08fce15138dc56889a5dcf/procedural/processor/src/main/java/gov/nasa/ammos/aerie/procedural/processor/ProcedureProcessor.java#L269) to use `ElementKind.RECORD_COMPONENT` instead of `FIELD` but would like to verify this further.

--- a/procedural/processor/src/main/java/gov/nasa/ammos/aerie/procedural/processor/ProcedureProcessor.java
+++ b/procedural/processor/src/main/java/gov/nasa/ammos/aerie/procedural/processor/ProcedureProcessor.java
@@ -266,7 +266,7 @@ public final class ProcedureProcessor implements Processor {
   private List<ParameterRecord> getExportParameters(final TypeElement exportTypeElement)
   {
     return exportTypeElement.getEnclosedElements().stream()
-                            .filter(e -> e.getKind() == ElementKind.FIELD) // Element must be a field
+                            .filter(e -> e.getKind() == ElementKind.RECORD_COMPONENT) // Element must be a record component
                             .map(e -> new ParameterRecord(e.getSimpleName().toString(), e.asType(), e))
                             .toList();
   }

